gobblin-config-management/gobblin-config-core/src/main/java/org/apache/gobblin/config/store/zip/ZipFileConfigStore.java

```diff
@@ -193,7 +194,7 @@ public Config getOwnConfig(ConfigKeyPath configKey, String version) throws Versi
   /**
    * Get path object using zipped file system and relative path
    */
-  private Path getDatasetDirForKey(ConfigKeyPath configKey) throws VersionDoesNotExistException {
-    return this.fs.getPath(this.storePrefix, configKey.getAbsolutePathString());
+  private Path getDatasetDirForKey(ConfigKeyPath configKey, String version) throws VersionDoesNotExistException {
```

Comment:
What does changing this to include the version do?

Reply:
The old implementation doesn't contain version in the signature which is not that correct, since the standard format of a config-store node should be(physical path):  _CONFIG_STORE/<versionNumber>/<NodePath>/main.conf. 

If the version number is missing, the resolved physical path is incorrect. The original implementation would work for the old unit test, because the original `zipConfigStore.zip` file doesn't have this version number layer which is not accurate according to Config-store's contract of file hierarchy.